config/routes.rb

```diff
@@ -129,6 +129,8 @@
     get "/" => "pages#index"
     get "dashboard" => "dashboard#index"
 
+    resources :invitations, only: [:destroy]
```

Comment:
Below this we have ` resources :invitations, only: [:new, :create, :destroy]`. I'd rather not duplicate that and just keep the original one. This also means you don't need to do the `skip_before_action`, because the team_id will be in the url.

Reply:
I'm generally a fan of shallow routing in order to clearly see the intention behind the URL. If we're going to destroy an invitation, we wouldn't need to the team ID.

Reply:
Also, the `skip_before_action` was there because I needed to put it in. Personally, I feel that the `TeamsController` has too many responsibilities. It retrieves the teams, at the same time provides the layout for the teams page, but there are routes for the teams pages that don't really require a team, such as joining one. There should be a separate controller for each responsibility.

Reply:
> I'm generally a fan of shallow routing in order to clearly see the intention behind the URL. If we're going to destroy an invitation, we wouldn't need the team ID.

OK - that makes sense. 

> Personally, I feel that the TeamsController has too many responsibilities. It retrieves the teams, at the same time provides the layout for the teams page, but there are routes for the teams pages that don't really require a team, such as joining one. 

I'm presuming you're talking about the `::TeamsController`, not the `Teams::TeamsController`? Yeah - I see your point here too. How would you structure it differently? Move `find_team` into `Teams::TeamsController` and have all the other controllers (e.g. `Teams::SolutionsController` inherit from `Teams::TeamsController` instead? That would make sense as a change to me.